### PR TITLE
Add C++ Pistache petstore server to Travis CI

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.1.x`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by wing328** [comment]
I got the following error in Travis:
```
Scanning dependencies of target pet_api_server
[  3%] Building CXX object CMakeFiles/pet_api_server.dir/PetApiMainServer.cpp.o
In file included from /home/travis/build/OpenAPITools/openapi-generator/samples/server/petstore/cpp-pistache/model/ApiResponse.h:22:0,
                 from /home/travis/build/OpenAPITools/openapi-generator/samples/server/petstore/cpp-pistache/api/PetApi.h:29,
                 from /home/travis/build/OpenAPITools/openapi-generator/samples/server/petstore/cpp-pistache/impl/PetApiImpl.h:28,
                 from /home/travis/build/OpenAPITools/openapi-generator/samples/server/petstore/cpp-pistache/PetApiMainServer.cpp:16:
/home/travis/build/OpenAPITools/openapi-generator/samples/server/petstore/cpp-pistache/model/ModelBase.h:22:20: fatal error: json.hpp: No such file or directory
 #include "json.hpp"
                    ^
compilation terminated.
make[2]: *** [CMakeFiles/pet_api_server.dir/PetApiMainServer.cpp.o] Error 1
make[1]: *** [CMakeFiles/pet_api_server.dir/all] Error 2
make: *** [all] Error 2
```

Do I need to install https://github.com/nlohmann/json manually?

cc  @ravinikam @stkrwork @fvarose @etherealjoy @martindelille

**Comment by wing328** [comment]
Also cc @CyrilleBenard as we discussed this yesterday.

**Comment by etherealjoy** [comment]
@wing328 
Missing nlohmann::json

#495 with the option below will solve all these automatically
```
--additional-properties addExternalLibs=true
```

File: samples/server/petstore/cpp-pistache/api/UserApi.cpp [hunk 1]
```diff
@@ -78,11 +75,11 @@ void UserApi::create_user_handler(const Pistache::Rest::Request &request, Pistac
 void UserApi::create_users_with_array_input_handler(const Pistache::Rest::Request &request, Pistache::Http::ResponseWriter response) {
 
     // Getting the body param
-    std::vector<User> user;
+    std::vector user;
```
Comment (etherealjoy) [inline samples/server/petstore/cpp-pistache/api/UserApi.cpp]: This is strange, because this new change here will not compile.

Comment (wing328) [reply samples/server/petstore/cpp-pistache/api/UserApi.cpp]: I've regenerated the samples and this issue seems to be gone.

**Comment by wing328** [comment]
@etherealjoy thanks for reviewing the change. The Travis now covers C++ Pistache Petstore server.

**Comment by wing328** [comment]
After merging the PR into master, the Travis CI jobs keep timing out:
```
The job exceeded the maximum time limit for jobs, and has been terminated.
```

I'll try to move it to other CI (e.g. CircleCI)

**Pull request merged**